### PR TITLE
Introduce WcsNDMap.cutout_and_mask_region

### PR DESCRIPTION
**Pull request by adonath**
**Description**

This pull request introduces a little refactoring to expose a step in `to_region_nd_map()`. The method is called `.cutout_and_mask_region()` and it return a cutout if the map, corresponding to the size of the region along with a mask, that defines the region. I refactored the existing `.to_region_nd_map()` method to use it. So testing is already done however I could maybe add one more unit test.

**Dear reviewer**

### CODE REVIEW

**Comment by adonath** [comment]
@MRegeard I think even if we mostly use it within Gammapy I think it should be a public method, because we will still call it from a different namespace. What I mean by that is methods are typically private when just called from the same namespace. So `Map._cutout_and_mask_region()` should only be called from other `Map` methods. If we call e.g. `counts._cutout_and_mask_region()` in other methods in Gammapy static code analyzers often complain.

File: gammapy/maps/wcs/ndmap.py [hunk 1]
```diff
@@ -529,6 +522,33 @@ def _plot_format_allsky(self, ax):
         lat.grid(alpha=0.2, linestyle="solid", color="w")
         return ax
 
+    def cutout_and_mask_region(self, region):
```
Comment (adonath) [inline gammapy/maps/wcs/ndmap.py:525]:
```suggestion
    def cutout_and_mask_region(self, region=None):
```

**Comment by MRegeard** [comment]
@adonath ok I see what you mean.

File: gammapy/maps/wcs/ndmap.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (adonath) [reply gammapy/maps/wcs/ndmap.py]: Done

**Review by registerrier — APPROVED** [review]
Thanks @adonath . No comment from my side.

**Review by MRegeard — APPROVED** [review]
Thanks @adonath, seems good to me !

**Pull request merged**